### DISCUSSION TITLE
spaCy doc creation and entity recognition

### DISCUSSION THREAD

**Discussion by Frazer-Nyambe** [explosion/spaCy#11719] [Help: Coding & Implementations] [answered]
i have created a pandas dataframe, i'm currently facing a problem on how to extract specific strings from the dataframe. given that the dataframe below is my first row with index "0", from the pandas dataframe, how do i convert the pandas dataframe to spaCy document so that i can extract "PERSON" entities for each column from the dataframe.

index | 1.txt | 2.txt |
-- | -- | -- |
0 | ACKNOWLEDGEMENT This dissertation would not have been possible without the support of many individuals and departments. My special thanks go to the following; 1. Dr. Paul Kelly for invaluable practical guidance and encouragement before and during the study. 2. The TROPGAN- Zambia under Dr. Kelly for providing the CMV -PCR kit, the TROPGAN laboratory for storage, processing and some financial assistance. 3. The TROPGAN- Zambia group for the contribution they made to the success of this study. I would like to specifically thank Cynthia Phiri for her tireless effort in processing the samples . 4. Pathology Department for the assistance they provided by doing post -mortem and getting the liver and serum samples. Dr Victor Mudenda , the late Dr. Joseph Banda, Dr Dennis Musonda and Mr Rabson Mwanza were invaluable. 5. Medicine Department for the support and the critique they provided before and during the study. Doctors in this department identified and provided the cases and cont rols for the study as well. 6. The members of staff under UTH who were doing locum under the study in collecting data. 7. My family and friends for encouragement they provided. 8. Finally I would like to thank the next -of-kins who allowed their beloved relatives to participate in the study. | iv ACKNOWLEDGEMENTS This study programme may no t have been possible if a num ber of people in their different capacities did not help in different ways . My greatest appreciation goes to Dr C harles Mwendabai N amafe of the University of Zambia , School of Education for his tireless corrections, help, and guidance . As my supervisor I would like to appreciate all the tireless work you put in to ensure that this study becomes a success. Thank you very much for all the criticisms. They have helped in the success of this study. I would also like to thank Mr. Kasonde Mundende of “Nkrumah University College of Education” , Geography section, for the support, knowledge and guidance he gave to me from the selection of a research topic to editing of my work. I am grateful for his parental advice and attention even during the times when he was ver y busy with his schedules. I also wish to thank Mr. Teddy Chibwe, Head of Mathematics Department at David Kaunda National Technical High School for his interest in guiding and editing my work amidst his busy schedules. Thank you very much. Further thanks go to Mr . Adrian Phiri of Mulungushi University for his help on Environmental Education information and guidance on how to w

**Comment by ljvmiranda921** [comment, marked as answer]
Hi @Frazer-Nyambe ,

You need to iterate over your pandas columns first, maybe by using [df.iterrows](https://pandas.pydata.org/docs/reference/api/pandas.DataFrame.iterrows.html) or [df.items](https://pandas.pydata.org/docs/reference/api/pandas.DataFrame.items.html). Also, if you already have the raw text, you don't have to convert it into a DataFrame. You can work with the texts as a string right away.

To convert it into a spaCy Doc. you can pass it to a pipeline. You also need to provide a model (e.g. `en_core_web_lg`) like so:

```python
import spacy

nlp = spacy.load("en_core_web_lg")
for text in texts:
    doc = nlp.pipe(text)
```

I highly-recommend looking at the [getting started](https://spacy.io/usage/spacy-101) and [API documentation](https://spacy.io/api/doc) for more info.
